content/knowledge_base/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,44 @@
+
+The protocol maintains the set of all known transactions using a dynamic, append-only Directed Acyclic Graph (DAG) structure. (TODO: add reference)
+There are two main positives from this:
+* DAG helps streamlining the path on which there are no conflicting transactions - a single vote on a DAG vertex implicitly votes for all transactions on the path to the genesis vertex
+* using the same principle as BTC, the DAG tree renders past decisions on the path of transactions as more and more difficult to undo without approval of the correct nodes
+Genesis block concept is implemented with the DAG having a single sink - "genesis vertex".
+
+When a new transaction is created its parents are defined in the DAG structure as its ancestors, forming the edges with their newly created decendant. The parent-child relationship in DAG is not mandatory for a child transaction may not have any relation to its parents transactions outputs. It can actually spend funds received in another transactions from its "ancestor set". The "ancestor set" is the set of all transactions reachable via parent edges back in history (TODO: add reference to the quote). Oppositely - the term "progeny" means all existing (and potentially existing) children transactions and their children.
+
+Main purpose of a consensus protocol is to avoid inclusion of conflicting transactions into the ledger. Even though a conflicting transaction characteristics are defined on an application level, the notion of conflict can be abstracted in order to define the conflicting set. In Avalanche every transaction belongs to a conflict set, meaning every transaction is a part of a set of transactions that contains one (if the set is a singleton set - the conflicting transactions will be potentially created or the singleton set contains a single virtuos transaction) or more transactions that are invalid towards each other existence.
+Following the purpose of reaching a consensus - only one transaction from any conflict set can be added to the DAG tree.
+
+Avalanche instantiates a Snowball instance for each conflict set. Avalanche treats the concept of repeated queries and multiple counters from Snowball taking advantage of DAG structure.Specifically, when a transaction T is queried, all of its ancestors reachable through the DAG edges from it are implicitly part of the query. Consequently, when a node responds positively to query of T only if T and all of its ancestors are the preferred option in their respective conflict sets.
+If more than a threshold of responders vote positively, the transactions gets a 'chit'. (TODO: add reference)
```

Comment:
```suggestion
Avalanche instantiates a Snowball instance for each conflict set.
Avalanche treats the concept of repeated queries and multiple counters from Snowball taking advantage of DAG structure.
Specifically, when a transaction `T` is queried, all of its ancestors reachable through the DAG edges from it are implicitly part of the query.
Consequently, nodes would respond positively to a query for transaction `T` only if `T` and all of its ancestors are the preferred option in their respective conflict sets.
If more than a given threshold of responders vote positively, the transactions gets a _chit_. (TODO: add reference)
```

Reply:
Updated the sentences + added references https://github.com/optakt/optakt.github.io/pull/13/commits/0a1cb902f3417c3bc20f9197df71d3d65a1b6421#diff-69f45991c0732c6f5ff297a5a86ffa91662d50369fc8057b3875361f5b80a2fcR36-R42
Thanks!